lib/net.js

```diff
@@ -825,7 +825,17 @@ protoGetter('bytesWritten', function bytesWritten() {
       bytes += Buffer.byteLength(el.chunk, el.encoding);
   });
 
-  if (data) {
+  if (Array.isArray(data)) {
+    // was a writev, iterate over chunks to get total length
+    for (var i = 0; i < data.length; i++) {
+      const chunk = data[i];
+
+      if (chunk instanceof Buffer)
+        bytes += chunk.length;
+      else
+        bytes += Buffer.byteLength(chunk.chunk, chunk.encoding);
+    }
+  } else if (data) {
```

Comment:
Perhaps this can be simplified a bit by merging with the if/else below:

```js
} else if (data instanceof Buffer) {
  // ...
} else {
  // ...
}
```